### DISCUSSION TITLE
any benchmark in speed and bytes allocation compared with datadog/zstd etc?

### DISCUSSION THREAD

**Discussion by kolinfluence** [klauspost/compress#690] [Q&A] [answered]
nice work!
any benchmark in speed and bytes allocation compared with datadog/zstd etc?

i did see it from badgerdb etc, seems truly random data, datadog is faster but lesser random this is faster? right?

**Comment by klauspost** [comment, marked as answer]
Both encoding and decoding buffers are 0 allocs/op, if you provide buffers that can hold the output and you re-use the Encoder/Decoder.

Haven't really bothered comparing, but I seem to remember it allocates on each call.
